static/src/js/components/GranuleImage/GranuleImage.js

```diff
@@ -13,128 +13,99 @@ import './GranuleImage.scss'
  * @param {Object} props - The props passed into the component.
  * @param {String} imageSrc - The image source.
  */
-class GranuleImage extends Component {
-  constructor(props) {
-    super(props)
 
-    this.state = {
-      isOpen: true,
-      isLoaded: false,
-      isLoading: true,
-      isErrored: false
-    }
-
-    this.handleToggleImage = this.handleToggleImage.bind(this)
-    this.handleImageLoaded = this.handleImageLoaded.bind(this)
-    this.handleImageErrored = this.handleImageErrored.bind(this)
-  }
-
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    const { imageSrc } = this.props
-    const { imageSrc: nextImageSrc } = nextProps
-    if (nextImageSrc && imageSrc !== nextImageSrc) {
-      this.setState({
-        isLoaded: false,
-        isLoading: true,
-        isErrored: false
-      })
-    }
-  }
-
-  handleToggleImage() {
-    const {
-      isOpen
-    } = this.state
-
-    this.setState({
-      isOpen: !isOpen
-    })
-  }
-
-  handleImageLoaded() {
-    this.setState({
-      isLoaded: true,
-      isLoading: false,
-      isErrored: false
-    })
-  }
-
-  handleImageErrored() {
-    this.setState({
-      isLoaded: false,
-      isLoading: false,
-      isErrored: true
-    })
+export const GranuleImage = ({
+  imageSrc
+}) => {
+  const [isOpen, setIsOpen] = useState(true)
+  const [isLoaded, setIsLoaded] = useState(false)
+  const [isLoading, setIsLoading] = useState(true)
+  const [isErrored, setIsErrored] = useState(false)
+
+  const containerClassName = classNames({
+    'granule-image': true,
+    'granule-image--is-open': isOpen
+  })
+
+  const imageClassName = classNames({
+    'granule-image__image': true,
+    'granule-image__image--is-loaded': isLoaded
+  })
+
+  const handleToggleImage = (() => {
+    setIsOpen(!isOpen)
+  })
+
+  const handleImageLoaded = (() => {
+    setIsLoaded(true)
+    setIsLoading(false)
+    setIsErrored(false)
+  })
+
+  const handleImageErrored = (() => {
+    setIsLoaded(false)
+    setIsLoading(false)
+    setIsErrored(true)
+  })
+
+  const loadImage = () => {
```

Comment:
`loadImage` is a confusing name, `handleImageLoading` might make more sense, and follows the other function names above. 

But it looks like the only time this function is called is within the useEffect, can you just call the 3 state updates within the useEffect?